src/test/ui/deref-suggestion.stderr

```diff
@@ -0,0 +1,32 @@
+error[E0308]: mismatched types
+  --> $DIR/deref-suggestion.rs:16:9
+   |
+16 |     foo(r_s);
+   |         ^^^ expected struct `std::string::String`, found reference
+   |
+   = note: expected type `std::string::String`
+              found type `&std::string::String`
+   = help: try with `*r_s`
```

Comment:
so I think that the overall code is fine -- if a bit restrictive -- but this is kind of a goofy suggestion, right? i.e., it's going to lead to a move error. Perhaps we should check if the type is copy before offering it?